dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/navigation/NavTool.java

```diff
@@ -142,8 +148,15 @@ protected NavResultHydrated getNav(Host host, String path, long languageId, User
                     children.add(nav);
                 } else if (item instanceof IHTMLPage) {
                     IHTMLPage itemPage = (IHTMLPage) item;
+                    boolean pageExistInRequestedLanguage = true;
 
-                    if (itemPage.getLanguageId() == languageId || LanguageWebAPI.canDefaultPageToDefaultLanguage()) {
+                    if(itemPage.getLanguageId() != languageId){
+                        pageExistInRequestedLanguage = findHTMLPageInRequestedLanguage(menuItems,itemPage.getIdentifier(),languageId);
+                    }
+
+
+                    if (itemPage.getLanguageId() == languageId ||
+                            (LanguageWebAPI.canDefaultPageToDefaultLanguage() && itemPage.getLanguageId() == defaultLanguage && !pageExistInRequestedLanguage)) {
```

Comment:
This is a very long second part of the OR. I would extract to a little method called `shouldAddPageInGivenLanguage(langId)`